corehq/ex-submodules/pillowtop/pillow/interface.py

```diff
@@ -176,10 +194,10 @@ def process_offset_chunk(chunk, context):
                         self._update_checkpoint(change, context)
                 else:
                     self._update_checkpoint(None, None)
+            context.flush_checkpoint_on_next_opportunity()
             process_offset_chunk(changes_chunk, context)
         except PillowtopCheckpointReset:
             process_offset_chunk(changes_chunk, context)
```

Comment:
I know you didn't change this, but it seems strange that `process_offset_chunk()` may raise `PillowtopCheckpointReset`, and then is called in the `except` block with no error handling—is that error expected to bubble up when called there?.

Reply:
I think everything you said is true; I hadn't originally noticed that this gets raised inside `process_offset_chunk`, but I think it does. That said, I don't have a good enough handle on the behavior here and what's expected to happen. I think in the worst case it just crashes and the supervisor restarts it, but I think if we were going to start messing with this we'd have to spec out the behavior we want and do more testing than just eyeballing it. @snopoke not sure if you have insights here. (I don't think it's blocking this PR, but I'm also curious.)